index.mjs

```diff
@@ -19,24 +32,25 @@ export default class Trouter {
 	use(route, ...fns) {
 		let handlers = [].concat.apply([], fns);
 		let { keys, pattern } = parse(route, true);
-		this.routes.push({ keys, pattern, method:'', handlers });
+		this.routes.push({ keys, pattern, method: '', handlers, mthidx: mthMap[''] });
 		return this;
 	}
 
 	add(method, route, ...fns) {
 		let { keys, pattern } = parse(route);
 		let handlers = [].concat.apply([], fns);
-		this.routes.push({ keys, pattern, method, handlers });
+		this.routes.push({ keys, pattern, method, handlers, mthidx: mthMap[method] });
 		return this;
 	}
 
 	find(method, url) {
-		let isHEAD=(method === 'HEAD');
+		let mthidx = mthMap[method];
+		let isHEAD = (mthidx === 2);
 		let i=0, j=0, k, tmp, arr=this.routes;
 		let matches=[], params={}, handlers=[];
 		for (; i < arr.length; i++) {
 			tmp = arr[i];
-			if (tmp.method.length === 0 || tmp.method === method || isHEAD && tmp.method === 'GET') {
+			if (tmp.mthidx === mthidx  || !tmp.mthidx || (isHEAD && tmp.mthidx===1) ) {
```

Comment:
The `!tmp.mthidx` is a bug. If a method that isnt in the map is added, it'll have `undefined` index which incorrectly matches this condition.

Reply:
hum I'm not sure how to rewrite it in a way you don't lose the perf you gain

Reply:
`tmp.mthidx === 0`